Arriba/Arriba/Model/Table.cs

```diff
@@ -354,13 +354,27 @@ public void AddColumnsFromBlock(DataBlock values)
         /// <param name="values">Set of Columns and values to add or update</param>
         public void AddOrUpdate(DataBlock values)
         {
+            AddOrUpdateOptions options = new Model.AddOrUpdateOptions();
+            options.AddMissingColumns = ShouldAddColumnsDynamically;
```

Comment:
Good idea - fixed. There's a risk we'll need to update other callers who currently expect this behavior, but I don't think there are any.